lib/protobuf/field/base_field.rb

```diff
@@ -122,10 +124,10 @@ def required?
 
       # FIXME: need to cleanup (rename) this warthog of a method.
       def set(message_instance, bytes)
-        return message_instance.__send__(setter, decode(bytes)) unless repeated?
-        return message_instance.__send__(getter) << decode(bytes) unless packed?
+        return message_instance[name] = decode(bytes) unless repeated?
```

Comment:
The `[]=` method now is the setter and `[]` is now the getter, so it should be the same. The old setter and getter still exist for runtime access of the proto, but they alias to the former two methods now.

For just a little more background, `[]` and `[]=` used to alias to the setter and getter instead, but we switched that around so that we didn't have this weird dynamic usage of the setters and getters.